core/src/main/java/org/elasticsearch/script/StoredScriptSource.java

```diff
@@ -266,8 +262,7 @@ public static StoredScriptSource parse(String lang, BytesReference content, XCon
                         //this is really for search templates, that need to be converted to json format
                         try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
                             builder.copyCurrentStructure(parser);
-                            return new StoredScriptSource(lang, builder.string(),
-                                Collections.singletonMap(Script.CONTENT_TYPE_OPTION, XContentType.JSON.mediaType()));
+                            return new StoredScriptSource(lang, builder.string(), Collections.emptyMap());
```

Comment:
I agree with @javanna that we should ultimately have different paths for user input and loading cluster state, but I think that can be done in a follow up PR.

I added the ability add compiler options because it may be needed for Painless in the future and there was already a single option there for templates anyway so it made sense to encapsulate it.  Content-type is reserved internally for templates only, so this check was preventing non-template scripts from specifying content-type.  The search template API never allows options to be input anyway so it's not a problem there.  Removal of this check should be safe for now since scripting languages at worst will ignore compiler options or in the case of Painless see an unrecognized option and fail at compile time.